src/fileSystemLoader.ts

```diff
@@ -21,16 +22,26 @@ const traceKeySorter = ( a, b ) => {
   }
 };
 
+export type Dictionary<T> = {
+  [key: string]: T | undefined;
+};
+
 export default class FileSystemLoader {
-  constructor( root, plugins ) {
+  private root: string;
+  private sources: Dictionary<string>;
```

Comment:
`Record<string, string>`

Reply:
this is not exactly the same, as

1. it doesn't account for `undefined`. though `Record<string, string | undefined>` would.
2. i'm not sure it this would be the semantic correct way to use `Record`. as far as i understand `Record` should be used with a subset of `string` unions, like in the following example, to express an object with a defined set of keys, that can be deduced at compile-time through the type system.

` type ThreeStringProps = Record<'prop1' | 'prop2' | 'prop3', string>`

(example from https://www.typescriptlang.org/docs/handbook/advanced-types.html)

whereas index access would better cover a dictonary like case, where the keys are not known at compile time.

> ... you could do something like Record<string, T>, but the idiomatic way of doing that is to use an index signature like { [k: string]: T }.

(from StackOverflow https://stackoverflow.com/questions/51936369/what-is-the-record-type-in-typescript)